### ISSUE TITLE
Computing CL-motif scores - error

### ISSUE THREAD

**Issue by skrakau** [skrakau/PureCLIP#19]
Hi Sabrina,  Managed to solve the issue.  It was something to do with the .fa file used.  Seems fine now and I ran pure clip Lon our eCLIP data.  Another issue came up though when I tried to use the CL-motifs for bias correction.  When I run compute_CLmotif_scores.sh with your motif files downloaded from GitHub, it loads the motifs in the file but crashes at the end.

nickwebster@Nicks-iMac-Pro Eclipse_results % compute_CLmotif_scores.sh mm10.fa Q0013_IP1_1_S15_L006_R1_001.adapterTrim.round2.rmRep.sorted.rmDup.sorted.bam motifs.xml motifs.txt fimo_clmotif_occurences.bed               
Using temporary directory '/var/folders/v9/tzx_rfsn28v2znvg94csf24m0000gn/T/tmp.2sRVeCSP'
Load reference file ...
Load candidate sites from BED file ...
Extract sequence for windows around candidate sites ...
Run FIMO ...
Using motif +TTTTNV of width 6.
Computing q-values.
Estimating pi_0 from a uniformly sampled set of 10000 p-values.
Estimating pi_0.
Estimated pi_0=0.961023
Using motif +TCTTTV of width 6.
Computing q-values.
Estimating pi_0 from a uniformly sampled set of 10000 p-values.
Estimating pi_0.
Estimated pi_0=0.958844
Using motif +TTTTWT of width 6.
Computing q-values.
Estimating pi_0 from a uniformly sampled set of 10000 p-values.
Estimating pi_0.
Estimated pi_0=0.943
Using motif +TBATCA of width 6.
Computing q-values.
Estimating pi_0 from a uniformly sampled set of 10000 p-values.
Estimating pi_0.
Estimated pi_0=0.984625
awk: can't open file BEGIN{FS=OFS="\t"} {print $2, NR};
 source line number 1

I looked at the sh script and it seems to be at the final step:

# replace motifs by its ID
# get DREME motifs
awk 'BEGIN{FS="\t| "; OFS="\t"} $1 == "MOTIF" {print $1, $2};' "$MOTIFS_TXT" > "$TEMP_DIR/dreme_motifs.txt"
# add id
awk -v n=$(wc -l < "$TEMP_DIR/dreme_motifs.txt") 'BEGIN{FS=OFS="\t"} {print $2, NR};' "$TEMP_DIR/dreme_motifs.txt" > "$TEMP_DIR/dreme_motifs.id.txt"
# replace motif with its id
awk 'BEGIN{FS=OFS="\t"} NR==FNR{a[$1]=$2} NR>FNR{$4=a[$4];print};' "$TEMP_DIR/dreme_motifs.id.txt" "$TEMP_DIR/FIMO_CL_MOTIFS/fimo.origPos.sn.u.txt" > "$BED_OUT"

Any thoughts what might be causing it?

Thanks, Nick

_Originally posted by @nick-webster in https://github.com/skrakau/PureCLIP/issues/18#issuecomment-656869954_

**Comment by skrakau** [comment]
https://github.com/skrakau/PureCLIP/issues/18#issuecomment-659428799

**Comment by skrakau** [comment]
Hi Nick, 

you are right, it might likely be caused by the meme version. At least for the versions > 4.11.2 there were frequent changes in the output format with newer versions. I used version 4.11.3, but also tested and handled the formats of versions 4.11.4 - 5.0.1 within the `compute_CLmotif_scores.sh` script , but not version 4.11.2 which you used. Did you update it already?

Best,
Sabrina

**Comment by nick-webster** [comment]
I updated to 5.1.1 but ran into the same problem.  I can try 4.11.3

Nick

Nick Webster
Sent from my iPhone.

On Jul 18, 2020, at 2:55 AM, Sabrina Krakau <notifications@github.com> wrote:

﻿

Hi Nick,

you are right, it might likely be caused by the meme version. At least for the versions > 4.11.2 there were frequent changes in the output format with newer versions. I used version 4.11.3, but also tested and handled the formats of versions 4.11.4 - 5.0.1 within the compute_CLmotif_scores.sh script , but not version 4.11.2 which you used. Did you update it already?

Best,
Sabrina
